server/gui/main.py

```diff
@@ -26,31 +27,33 @@ def __init__(self):
         super(MainWindow, self).__init__(None)
         self.cef_widget = None
         self.data_widget = None
-        self.parent_conn, self.child_conn = Pipe()
-        self.load_emitter = Emitter(self.parent_conn, WorkerSignals)
-        self.load_emitter.signals.error.connect(self.restartOnError)
-        self.emitter_thread = threading.Thread(target=self.load_emitter.run, daemon=True)
-        self.emitter_thread.start()
+        self.parent_conn, self.child_conn = None, None
+        self.load_emitter = None
+        self.emitter_thread = None
         self.worker = None
         self.url = f"http://localhost:{GUI_PORT}/"
         self.setWindowTitle("cellxgene")
 
         # Strong focus - accepts focus by tab & click
         self.setFocusPolicy(Qt.StrongFocus)
         self.setupLayout()
-        # self.setupMenu()
+        self.setupMenu()
 
-    def restartOnError(self):
-        if self.worker:
-            self.terminateWorker()
-        self.parent_conn.close()
+    def setupServer(self):
+        self.shutdownServer()
         # close emitter on error/finished
         self.parent_conn, self.child_conn = Pipe()
         self.load_emitter = Emitter(self.parent_conn, WorkerSignals)
         self.emitter_thread = threading.Thread(target=self.load_emitter.run, daemon=True)
         self.emitter_thread.start()
         # send to load with error message?
 
+    def shutdownServer(self):
+        if self.worker:
+            self.terminateWorker()
+        if self.parent_conn:
+            self.parent_conn.close()
+
     def terminateWorker(self):
```

Comment:
Question: why call Process.close()?  My understanding from the 3.7 docs is that it will only succeed if the underlying process has already exited.    The `terminate()` function seems to also release resources, so couldn't you replace this entire function with a call to `self.worker.terminate()` and have the same effect?

Reply:
Fixed on the other. You're right. Close was never being called.